qp_shotgun/kneaddata/kneaddata.py

```diff
@@ -187,37 +187,48 @@ def _run_commands(qclient, job_id, commands, msg):
     return True, ""
 
 
-def _per_sample_ainfo(out_dir, samples):
-    pf = []
+def _per_sample_ainfo(out_dir, samples, fwd_and_rev=False):
     ainfo = []
-    for rp, _, _, _ in samples:
-        smd = partial(join, out_dir, rp)
-
-        # matching forward/reverse
-        fname = smd('%s.trimmed.1.fastq' % rp)
-        if exists(fname):
-            pf.append((fname, 'preprocessed_fastq'))
-        fname = smd('%s.trimmed.2.fastq' % rp)
-        if exists(fname):
-            pf.append((fname, 'preprocessed_fastq'))
+
+    if fwd_and_rev:
+        paired, ur1, ur2 = [], [], []
+        for rp, _, _, _ in samples:
+            smd = partial(join, out_dir, rp)
+
+            # matching forward/reverse
+            fname = smd('%s_paired_1.fastq' % rp)
+            if exists(fname):
+                paired.append((fname, 'preprocessed_fastq'))
+            fname = smd('%s_paired_2.fastq' % rp)
+            if exists(fname):
+                paired.append((fname, 'preprocessed_fastq'))
+
+            # unmatching forward
+            fname = smd('%s_unmatched_1.fastq' % rp)
+            if exists(fname):
+                ur1.append((fname, 'preprocessed_fastq'))
```

Comment:
can we add the following code to ensure that there is a file there, even if it is empty?

```
if exists(fname):
    ur1.append((fname, 'preprocessed_fastq'))
else:
    with open(fname, 'a'):
        os.utime(fname, times)
```

This will help downstream when there are no unmatched fastqs, as in when we have very good sequencing runs (often don't have any unpaired R2) or small files.

will need for unmatched_1 and unmatched_2